apps/site/assets/ts/models/alert.ts

```diff
@@ -159,24 +159,33 @@ const isCurrentLifecycle = ({ lifecycle }: Alert): boolean =>
   lifecycle === "ongoing" ||
   lifecycle === "ongoing_upcoming";
 
-export const isCurrentAlert = (
+export const isBannerAlert = ({ effect }: Alert): boolean =>
+  ["suspension", "stop_closure", "station_closure", "shuttle"].includes(effect);
+
+export const isInNextXDays = (
   alert: Alert,
+  days: number,
   currentDate: Date = new Date()
 ): boolean => {
+  const xDays = add(Date.now(), { days });
+  xDays.setHours(23, 59, 59); // set to end of X day
   if (!alert.active_period) return false;
   const dateRanges = alert.active_period.map(ap => activePeriodToDates(ap));
   const isInARange = dateRanges.some((range): boolean => {
     const [start, end] = range;
-    if (!start || !isValid(start)) return false; // end might be null for ongoing alerts
-    return (
-      currentDate >= start && (end && isValid(end) ? currentDate <= end : true)
-    );
+    if (!start || !isValid(start)) return false; // start might be null for ongoing alerts
+    if (!end || !isValid(end)) return true; // end might be null for ongoing alerts
+
+    if (days === 0) {
+      return start <= currentDate && end >= currentDate;
+    }
+    return start <= xDays && end >= currentDate && end <= xDays;
```

Comment:
I don't think start can be null for ongoing alerts.  Based off the previous logic (and here) false is returned if there is not a start, so it shouldn't have a comment about being an ongoing alert if start is null.

I think the `currentDate` variable can be removed.  You can make days optional, and set it to 0 by default (maybe that is too confusing).  I say that because it looks like `currentDate` equals `xDays` when days is `0`

The logic at the end needs some rethinking.  It looks like this will return false if the alert ends after `xDays` even though it started before `xDays`. That doesn't seem right.  I think you just want to replication the `start <= xDays && end >= xDays`

Reply:
I agree this function defaulting to 0 days when no xDays isn't intuitive, will leave `xDays` as required (although saying 0 days from now == currently happening is also odd). 

Will return `return start <= xDays && end >= currentDate` since the end of the alert could be before xDays but after the current date